app/models/miq_worker.rb

```diff
@@ -135,11 +144,11 @@ def self.sync_workers
       _log.info("Workers are being synchronized: Current #: [#{current}], Desired #: [#{desired}]")
 
       if desired > current && enough_resource_to_start_worker?
-        (desired - current).times { result[:adds] << start_worker.pid }
+        (current..desired-1).each { |instance| result[:adds] << start_worker(:instance_name => instance).pid }
       elsif desired < current
         w = w.to_a
-        (current - desired).times do
-          ww = w.pop
+        (desired..current+1).times do
```

Comment:
I think you forgot to remote the .times here?  I'm not sure a Range#times is even a thing?

Reply:
Also `desired.downto(current)` for readability.

Reply:
I agree with @Fryguy... haha, just make sure there's no off by one errors here  😉